### ISSUE TITLE
"Cyrillic: Count io (ё) as e" in v1.4.0beta works not as espected

### ISSUE THREAD

**Issue by DmFedorov** [Predelnik/DSpellCheck#125]
I did not check this item earlier (in previous versions), but I think nothing has changed.

Actually, what's the problem?
The problem is that this Checkbox takes into account only the worst variant of "correct behavior"
Checkbox allows only one way of how a dictionary is made.
And this Checkbox (if it is correctly written) at the current moment means the following:
"Cyrillic: Count e as io (e)" (reverse sequence)

If a dictionary does not take into account the letter ё (in the dictionary there are no words with the letter ё), then when I put a letter ё (in the word of the text) instead of letter e, that word will not be underlined.
If there are several letters "е" in the word, I can put a few letters ё (which is generally unacceptable) or put it in the wrong place of the word - the word will not be underlined anyway.

Now imagine that I have a dictionary that takes into account the letter ё.
The correct Russian dictionary, such as it should be.
However, the practice of writing Russian words very often does not take into account the letter ё.
You can write the letter e and to all will be clear that it will sound like ё.
It is as though admissible. And for such cases it is not necessary to check the letter ё.
For such cases i need Checkbox: Cyrillic: Count io (ё) as e
If this does not have such an Checkbox, then I must additionally describe all cases with the letter e.
Model: A rule with the letter ё, and exactly the same rule with the letter e.
These are tens of kilobytes of rules.
Otherwise, words with the letter e instead of ё will be underlined.
So Checkbox "Cyrillic: Count io (ё) as e" is very useful.

But such a correct behavior does not work.
If the dictionary contains a word with the letter ё, this word will be underlined directly in the dictionary!!!
And all this because the letters ё as it does not exist, when I have today's Checkbox Cyrillic: Count io (ё) as e.

**Comment by Predelnik** [comment]
The sole and also historical reason for this checkbox existence is aspell dictionary which doesn't have any ё in it. And its meaning is very simple it transforms all ё into е before checking words against the dictionary and yes it loses some information possibly useful to spell-checking. 

Checkin ё-less texts against ё-ful dictionary could be useful of course, but it seems like it's not possible to do in a simple way. I either have to try to replace every е in the word with ё in any combinations before checking against the dictionary which is really clumsy or try to find the similar word ё-insesitively equal to the current word in suggestions but it's actually not a surefire way too -- I found that word `ее` doesn't have `её` in all of its 15 suggestion for Russian Hunspell dictionary. 

All in all, I do think the fact that Hunspell provides both ё-full and ё-less dictionaries should be enough for most of the usages to be honest without implementing these hacky workarounds. Native windows spell-checking Russian dictionary contains both ё-ified and un-ё-ified versions of the words but that of course could remove possibility to do the precise spell-checking of certain texts.

TLDR: This checkbox mostly exists only for Aspell and for historical reasons, I'm sorry if its meaning is unclear.

**Comment by Predelnik** [comment]
Since I don't see algorithmic solution to the problem described I'm closing this now.

I would be happy to improve this checkbox or add another one if how it works would be described algorithmically. Personally I have no idea, feel free to comment even if issue is closed.

Usefullness of this checkbox in its current state I hope was explained.
